### PR TITLE
Add missing Geany parsers

### PR DESCRIPTION
**Pull request by techee**
This pull request adds the extra parsers we have in Geany except those defined in c.c and except the markdown parser which is readLine() based and not really superior to the regex-based parser ctags uses (we should probably use the ctags parser in Geany instead).

The added parsers are:
* Abaqus (some obscure hardware definition language)
* Abc (https://en.wikipedia.org/wiki/ABC_notation)
* Docbook (https://en.wikipedia.org/wiki/DocBook)
* Haskell (https://en.wikipedia.org/wiki/Haskell_(programming_language))
* Haxe (https://en.wikipedia.org/wiki/Haxe)
* Nsis (https://en.wikipedia.org/wiki/Nullsoft_Scriptable_Install_System)
* Powershell (https://en.wikipedia.org/wiki/PowerShell)
* Txt2tags (https://en.wikipedia.org/wiki/Txt2tags)

For more details see the individual commits.

### CODE REVIEW

File: parsers/abc.c [hunk 1]
```diff
@@ -28,8 +28,8 @@
 */
 
 static kindDefinition AbcKinds[] = {
-	{ true, 'm', "member", "sections" },
-	{ true, 's', "struct",  "header1"}
+	{ true, 's', "section", "sections" },
+/*	{ true, 'h', "header",  "header1"} */
```
Comment (masatake) [inline parsers/abc.c]: I wonder why this one is commented out.

File: parsers/abaqus.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
+/*
+ *   This source code is released for free distribution under the terms of the
```
Comment (masatake) [inline parsers/abaqus.c]: Is it possible to write the copyright holder explicitly?

File: parsers/abc.c [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#include "read.h"
+#include "vstring.h"
+#include "routines.h"
+#include "entry.h"
+
+/*
+*   DATA DEFINITIONS
+*/
+
+static kindDefinition AbcKinds[] = {
```
Comment (masatake) [inline parsers/abc.c]: If 's' is not used, let's remove it.
Event there is only one kind, I would like you to define enum for kinds like:
```
enum AbcKind {
   K_MEMBER,
};
```

File: parsers/haskell.c [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+	def->kindCount  = ARRAY_SIZE(HaskellKinds);
+	def->extensions = extensions;
+	def->parser     = findNormalHaskellTags;
+	return def;
+}
+
+extern parserDefinition* LiterateHaskellParser (void)
+{
+	static const char *const extensions [] = { "lhs", NULL };
+	parserDefinition* def = parserNew ("Literate Haskell");
```
Comment (masatake) [inline parsers/haskell.c]: Let's avoid to use whitespace in a parser name.

File: parsers/abc.c [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+			return false;
+	}
+	return true;
+}*/
+
+
+static void makeAbcTag (const vString* const name, bool name_before)
+{
+	tagEntryInfo e;
+	initTagEntry (&e, vStringValue(name), 0);
```
Comment (masatake) [inline parsers/abc.c]: I would like you to use 'K_MEMBER' instead of 0.

**Comment by masatake** [comment]
I will add a selector for .hx file.

**Comment by masatake** [comment]
Thank you very much.

File: parsers/abc.c (hunk 3 above)
Comment (techee) [reply parsers/abc.c]: I've removed all the commented-out code from the parser which is unmaintained and there doesn't seem to be anything of a great value. I've also made some cleanups in the parser.

File: parsers/haskell.c (hunk 4 above)
Comment (techee) [reply parsers/haskell.c]: Done. By the way, this variant of the parser is unused right now (and we don't use it in Geany either, just the ordinary HaskellParser).

**Comment by techee** [comment]
> Could you add the pages about the languages in the commit log:
> 
> > 
> 
> ```
> Abaqus (some obscure hardware definition language)
> Abc (https://en.wikipedia.org/wiki/ABC_notation)
> Docbook (https://en.wikipedia.org/wiki/DocBook)
> Haskell (https://en.wikipedia.org/wiki/Haskell_(programming_language))
> Haxe (https://en.wikipedia.org/wiki/Haxe)
> Nsis (https://en.wikipedia.org/wiki/Nullsoft_Scriptable_Install_System)
> Powershell (https://en.wikipedia.org/wiki/PowerShell)
> Txt2tags (https://en.wikipedia.org/wiki/Txt2tags)
> ```
> 
> to each parser source file?

Done, including Abaqus as well (which actually isn't a hardware-definition language as I thought but some simulation software language).

File: docs/news.rst [hunk 6]
```diff
@@ -67,11 +72,13 @@ The following parsers have been added:
 * Maven2 *libxml*
 * Moose *perl basesd subperser*
 * M4
+* Nsis
 * ObjectiveC
 * Passwd *optlib*
 * PuppetManifest *optlib*
 * Perl6
 * Pod *optlib*
+* Powershell
```
Comment (masatake) [inline docs/news.rst]: Very minor issue. Powershell should be PowerShell.

Comment (techee) [reply docs/news.rst]: Fixed, including a few more capitalization problems.

Comment (masatake) [reply docs/news.rst]: Thank you.

**Comment by masatake** [comment]
If there is no trouble at Geany side, I would like you to do one of following two:
A. squash the commits into one then merge it, or
B. convert the commits into per-parser commits and merge them.

**Comment by techee** [comment]
>If there is no trouble at Geany side, I would like you to do one of following two:
>A. squash the commits into one then merge it, or
>B. convert the commits into per-parser commits and merge them.

No problem, I guess A will be easier to do.

**Comment by techee** [comment]
I opened a separate pull request with the rebased version in #2106 so the individual commits are still visible if e.g. @b4n wants to have a look at them. Closing this pull request.

**Pull request closed without merging**